src/main/java/io/openliberty/tools/common/plugins/util/InstallFeatureUtil.java

```diff
@@ -516,17 +597,28 @@ public void installFeatures(boolean isAcceptLicense, List<String> featuresToInst
                     throw new PluginExecutionException(exceptionMessage);
                 }
             }
-            Collection<File> artifacts = downloadEsas(resolvedFeatures);
+            Map<File, String> artifactsToExt = downloadEsas(resolvedFeatures, featureToExtMap);
+            Set<File> artifacts = artifactsToExt.keySet();
 
             StringBuilder installedFeaturesBuilder = new StringBuilder();
             Collection<String> actionReturnResult = new ArrayList<String>();
             for (File esaFile : artifacts) {
                 mapBasedInstallKernel.put("license.accept", acceptLicenseMapValue);
                 mapBasedInstallKernel.put("action.install", esaFile);
-                if (to != null) {
+                String ext = artifactsToExt.get(esaFile);
+                mapBasedInstallKernel.put("to.extension", TO_USER);
+                
+                if (ext!= null && ext != "" && to != null) {
```

Comment:
Should string comparison use the `equals` method?
`!ext.equals("")`
Same for line 614 below.